src/Transports.AspNetCore/AzureFunctions/AzureGraphQLActionResult.cs

```diff
@@ -0,0 +1,41 @@
+namespace GraphQL.Server.Transports.AspNetCore.AzureFunctions;
+
+/// <summary>
+/// An <see cref="IActionResult"/> that executes a GraphQL request for the specified schema.
+/// </summary>
+public class AzureGraphQLActionResult<TSchema> : IActionResult
+    where TSchema : ISchema
+{
+    private readonly HttpRequest _httpRequest;
+    private readonly IAzureGraphQLMiddleware<TSchema> _middleware;
+
+    /// <summary>
+    /// Initializes a new instance with the specified <see cref="HttpRequest"/>.
+    /// </summary>
+    /// <param name="httpRequest"></param>
+    public AzureGraphQLActionResult(HttpRequest httpRequest)
+    {
+        _httpRequest = httpRequest;
+        _middleware = httpRequest.HttpContext.RequestServices.GetRequiredService<IAzureGraphQLMiddleware<TSchema>>();
+    }
+
+    /// <inheritdoc/>
+    public virtual Task ExecuteResultAsync(ActionContext context)
+        => _middleware.InvokeAsync(_httpRequest, static httpContext =>
```

Comment:
Why do you need to pass `next` delagate at all?

Reply:
The `GraphQLHttpMiddleware` is designed around having a `next` delegate available if, for example, GET requests are disabled.  In this case the ActionResult returns a NotFound message, similar to the default behavior of ASP.NET Core.

Reply:
https://github.com/dotnet/aspnetcore/blob/1ef289a7589b3b095d0015b55b47aebd8e9667fd/src/Http/Http/src/Builder/ApplicationBuilder.cs#L135

Reply:
These are Azure Functions, and does not use the ASP.NET Core pipeline.  There is no IApplicationBuilder.

Reply:
Even within ASP.NET Core, the code you pointed to only occurs if the end of the pipeline is reached.  An `ActionResult` is an end of a pipeline, and ASP.NET Core considers the request to be handled at that point.  It would not reach the code you pointed to.

Reply:
Notice the signature: `Task ExecuteResultAsync(ActionContext context)`.  There is no `next` delegate.  If there were, this would contain the code you linked to.  But there isn't.  The code runs here, and ASP.NET Core or Azure Functions considers the request handled.  That is why while mapping onto our middleware, I provide the middleware with the code to set a 404 response -- in case for some reason the middleware doesn't handle the request, it will respond with 404.

Reply:
> These are Azure Functions, and does not use the ASP.NET Core pipeline. There is no IApplicationBuilder.

Well, it does have a number of dependencies on ASP.NET Core 2.x libraries.  I honestly do not know how much or how little it uses ASP.NET Core functionality.

Reply:
ok